src/tests/CompleteOptions.test.tsx

```diff
@@ -0,0 +1,71 @@
+import '@testing-library/jest-dom';
+
+import { h } from 'preact';
+import { clickPurposeCheckbox, getPurposeCheckState, render } from './utils';
+import { test, expect } from '@jest/globals';
+
+import { CompleteOptions } from '../components/CompleteOptions';
+import { fireEvent } from '@testing-library/preact';
+import { MOCK_PURPOSES } from './constants';
+import { DEFAULT_PURPOSE_TO_MESSAGE_KEY } from '../components/constants';
+import { sortByPurposeOrder } from '../helpers';
+
+test('CompleteOptions', () => {
+  const prevPurposes = { ...window.airgap?.getConsent().purposes };
+
+  const screen = render(<CompleteOptions handleSetViewState={() => null} />);
+  const { container, snapshot } = screen;
+
+  expect(snapshot).toMatchSnapshot();
```

Comment:
I'm not a huge fan of using snapshots, as they're pretty brittle and need extra logic to save. What was the primary purpose of including snapshots here? Was it just to check if some of the structure is correct? Could we potentially do that with a few spot checks instead to reduce brittleness?

Reply:
tbh i dont like them either, but the instigating event for these tests calls for providing some sort of barrier against reordering of purposes within the ui. thats not really going to be detected by a unit test unless we manually specify an ordering in every test, which seems about as bad as a snapshot to me (and its more manual). id otherwise be happy to delete this